src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryFormatterWriter.cs

```diff
@@ -112,23 +112,23 @@ internal void WriteSerializationHeader(int topId, int headerId, int minorVersion
             record.Write(this);
         }
 
-        internal void WriteObject(NameInfo nameInfo, NameInfo typeNameInfo, int numMembers, string[] memberNames, Type[] memberTypes, WriteObjectInfo[] memberObjectInfos)
+        internal void WriteObject(NameInfo nameInfo, NameInfo? typeNameInfo, int numMembers, string[] memberNames, Type[] memberTypes, WriteObjectInfo[] memberObjectInfos)
         {
             InternalWriteItemNull();
             int assemId;
             int objectId = (int)nameInfo._objectId;
 
-            string objectName = objectId < 0 ?
-                objectName = typeNameInfo.NIname : // Nested Object
-                objectName = nameInfo.NIname; // Non-Nested
+            string? objectName = objectId < 0 ?
+                typeNameInfo!.NIname : // Nested Objec
```

Comment:
`typeNameInfo` explicilty called with null once, but don't see if `objectId < 0` condition could ever happen so banged it for now, let me know what is better

PS: I see I mistakenly removed `t` from comment, will put back